src/integration-test/java/com/commercetools/sync/integration/inventories/InventorySyncIT.java

```diff
@@ -179,9 +149,7 @@ void sync_WithKeyToExistingSupplyChannelInPlaceOfReferenceId_ShouldUpdateInvento
          * Prepare InventoryEntryDraft of sku SKU_1 and reference to supply channel of key SUPPLY_CHANNEL_KEY_1.
          * Please note that the key is provided in place of Referenced id.
          */
-        final Reference<Channel> supplyChannelReference = Channel.referenceOfId(SUPPLY_CHANNEL_KEY_1);
-        assertThat(supplyChannelReference.getObj()).isNull();
-        assertThat(supplyChannelReference.getId()).isEqualTo(SUPPLY_CHANNEL_KEY_1);
+        final ResourceIdentifier<Channel> supplyChannelReference = ResourceIdentifier.ofId(SUPPLY_CHANNEL_KEY_1);
```

Comment:
why not `ResourceIdentifier.ofKey(SUPPLY_CHANNEL_KEY_1);` ?

Reply:
because this is the way we do reference resolution on the sync java library in general: We always provide the key in place of the id field (for consistency) since there still (will change in the future) exists references which are of type `Reference` not `ResourceIdentifier`. And in a `Reference` there is no `key` field, so we misuse the `id` field for it. 

However, this ought to be changed very soon.